code/modules/food_and_drinks/food/foods/ingredients.dm

```diff
@@ -140,6 +205,19 @@
 	filling_color = "#7D5F46"
 	list_reagents = list("nutriment" = 2, "sugar" = 2, "cocoa" = 2)
 
+///Chocolate crumbles/pile
+/obj/item/reagent_containers/food/snacks/chocolatebar/attackby(obj/item/I, mob/user, params)
+	if(istype(I, /obj/item/kitchen/knife))
+		if(isturf(loc))
+			new /obj/item/reagent_containers/food/snacks/choc_pile(loc)
+			to_chat(user, "<span class='notice'>You cut [src] into little crumbles.</span>")
+			qdel(src)
+		else
+			to_chat(user, "<span class='notice'>You need to put [src] on a surface to cut it out!</span>")
+	else
+		..()
```

Comment:
return the value.

Reply:
Could you be a little more specific? I don't know where I should put the return, as it wasn't in the example I used to base my code (The original dough and rolling pin code), I looked at it because maybe I missed it but isn't there neither

Reply:
@Fox-McCloud  It works perfectly as it is so It'd be good if you tell me if it's necessary to add it, why and how
I'm scared to do it wrong and break something